src/main/java/bio/terra/service/filedata/azure/tables/TableDirectoryDao.java

```diff
@@ -53,139 +62,184 @@
     }
     tableClient.deleteEntity(leafEntity);
 
     FireStoreDirectoryEntry leafEntry = FireStoreDirectoryEntry.fromTableEntity(leafEntity);
-    String datasetId = leafEntry.getDatasetId();
-    String lookupPath = fileMetadataUtils.makeLookupPath(leafEntry.getPath());
+    String lookupPath = FileMetadataUtils.makeLookupPath(leafEntry.getPath());
     while (!lookupPath.isEmpty()) {
       // If there are no entries that share this path as their directory path,
       // delete the directory entry
-      String filterPath = fileMetadataUtils.makePathFromLookupPath(lookupPath);
+      String filterPath = FileMetadataUtils.makePathFromLookupPath(lookupPath);
       ListEntitiesOptions options =
           new ListEntitiesOptions().setFilter(String.format("path eq '%s'", filterPath));
-      if (TableServiceClientUtils.tableHasEntries(tableServiceClient, TABLE_NAME, options)) {
+      if (TableServiceClientUtils.tableHasEntries(tableServiceClient, tableName, options)) {
         break;
       }
-      TableEntity entity = lookupByFilePath(tableServiceClient, datasetId, lookupPath);
+      TableEntity entity =
+          lookupByFilePath(tableServiceClient, collectionId, tableName, lookupPath);
       if (entity != null) {
         tableClient.deleteEntity(entity);
       }
-      lookupPath = fileMetadataUtils.getDirectoryPath(lookupPath);
+      lookupPath = FileMetadataUtils.getDirectoryPath(lookupPath);
     }
     return true;
   }
 
   // Each dataset/snapshot has its own set of tables, so we delete the entire directory entry table
-  public void deleteDirectoryEntriesFromCollection(TableServiceClient tableServiceClient) {
-    if (TableServiceClientUtils.tableExists(tableServiceClient, TABLE_NAME)) {
-      TableClient tableClient = tableServiceClient.getTableClient(TABLE_NAME);
+  public void deleteDirectoryEntriesFromCollection(
+      TableServiceClient tableServiceClient, String tableName) {
+    if (TableServiceClientUtils.tableExists(tableServiceClient, tableName)) {
+      TableClient tableClient = tableServiceClient.getTableClient(tableName);
       tableClient.deleteTable();
     } else {
       logger.warn(
           "No storage table {} found to be removed.  This should only happen when deleting a file-less dataset.",
-          TABLE_NAME);
+          tableName);
     }
   }
 
   // Returns null if not found - upper layers do any throwing
   public FireStoreDirectoryEntry retrieveById(
-      TableServiceClient tableServiceClient, String fileId) {
-    TableEntity entity = lookupByFileId(tableServiceClient, fileId);
+      TableServiceClient tableServiceClient, String tableName, String fileId) {
+    TableEntity entity = lookupByFileId(tableServiceClient, tableName, fileId);
     return Optional.ofNullable(entity)
         .map(d -> FireStoreDirectoryEntry.fromTableEntity(entity))
         .orElse(null);
   }
 
+  // returns empty list if no items are found
+  public List<FireStoreDirectoryEntry> batchRetrieveById(
```

Comment:
Do we still need `batchRetrieveById`? Looks like it's not being used anywhere

Reply:
Huh. Good catch - I can't remember why I thought we needed this.